common/unsafe/src/main/java/org/apache/spark/sql/catalyst/util/CollationAwareUTF8String.java

```diff
@@ -296,6 +296,28 @@ public static String toLowerCase(final String target, final int collationId) {
     return UCharacter.toLowerCase(locale, target);
   }
 
+  /**
+   * Convert the input string to titlecase using the ICU root locale rules.
+   *
+   * @param target the input string
```

Comment:
IMO, there is not much value in writing generic @param/@return comments. I am fine with just main header.